### DISCUSSION TITLE
Inline clozes integrated with SRS  for long text tiddlers

### DISCUSSION THREAD

**Discussion by arunnbabu81** [oflg/TiddlyMemo#5] [Ideas]
I had asked for help in the talk tiddlywiki forum (https://talk.tiddlywiki.org/t/help-needed-to-create-a-cloze-flashcard-type-by-combining-strex-macro-by-telmiger-and-macros-from-anwiki/4959) , but didn't get much responses there. Thats why I am asking for your help. 

As mentioned in the abovementioned post, is it possible to create inline clozes with SRS feature.

Clozes can be created by using telmigers strex macro- https://tid.li/tw5/hacks.html#TextStretch).

But I don't know how to integrate it with spaced repetition as in anwiki (https://anwiki.tiddlyspot.com/#Macros) or tiddlymemo. 

I have created a demo  here -https://arsheth-srs.tiddlyhost.com/

I think it must be possible as done in the sticky todo plug in by stobot (https://stobot.github.io/sticky/#%24%3A%2Fplugins%2Fstobot%2Fsticky%2Fmacros) and JanJo (https://szen.io/ToDo/#%24%3A%2Fplugins%2FJJ%2FToDo%2Fmacros)

I don't like creating seperate tiddlers to create flashcards. I want it to be inline within my notes as I described above. But I don't have the ability to do it right now. Also I don't have the time to wait till I acquire the skills to create it on my own. Thats why I am sharing my idea here. Is it possible to create such a system atleast as a subplug in?

**Comment by oflg** [comment]
Thank you for your interest in TiddlyMemo.

I probably wouldn't do a plugin like this in TiddlyMemo, indeed because a separate tiddler in TiddllyMemo is a cloze question.

Perhaps you could look again at [TWSR](https://tadeas.dev/twsr/), [TakeAway](https://groktiddlywiki.com/read/) and [TiddlyRemember](https://sobjornstad.github.io/TiddlyRemember/).

**Comment by oflg** [comment]
Hello @arunnbabu81.

 I have updated TiddlyMemod to v1.8.0 today. This version uses macros to create cloze and I don't know if that is what you need.

![image](https://user-images.githubusercontent.com/42492105/224480540-109b3d83-445a-495c-8e77-11c292c05c94.png)
![image](https://user-images.githubusercontent.com/42492105/224480580-f2380fa5-de46-4495-aab7-c5a0f8533a95.png)


See the documentation for details: https://tiddlymemo.org/manual/#How%20to%20use%20Cloze%20and%20Select%20Question%3F

**Reply by arunnbabu81** [reply]
Thank you @oflg for adding cloze macro....i will check it out when I am on my desktop and give you feedback
